keps/sig-auth/2784-csr-duration/README.md

```diff
@@ -0,0 +1,473 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Extend the Certificates API with a mechanism to allow clients to request a
+specific duration for the issued certificate.
+
+## Motivation
+
+Certificates issued through the certificate signing requests API are not
+revocable today.  Furthermore, clients lack a way to control the duration of the
+issued certificate.  Certificate approvers may have trust distinctions for
+different clients.  For example, if an approver knows that a client stores their
+private key in a HSM, they may be willing to approve for a longer certificate
+duration.  Signers lack a per-CSR mechanism to control the certificate duration,
+and thus are forced to learn this information out of band.  The built-in signers
+use the same duration for all issued certificates, with the default being a one
+year certificate lifespan (that is irrevocable without rotating the signer).
+
+To increase the overall security of the Kubernetes ecosystem and allow clients
+to request a particular duration for issued certificates, we propose extending
+the Kubernetes certificates signing request API with a new optional spec field
+that can be used to request a particular duration.
+
+### Goals
+
+- Allow clients to request a duration for requested certificates
+- Allow the CSR API to be easily [adopted in the Kubernetes ecosystem](https://github.com/jetstack/cert-manager/pull/3646)
+
+### Non-Goals
+
+- Requiring signers to honor the requested duration
```

Comment:
is configuring existing in-tree consumers of the CSR API (kubelets) to make use of this feature a non-goal?

Reply:
IMO, it is a non-goal but I could see part of the GA requirements being some kind of validation that the kubelet CSR flow could use this... which may require the config?  I would prefer to defer this wiring to the existing efforts for CSR rotation but honestly I cannot remember where we are with that.

Reply:
Added explicit non-goal.